lab3/lab3/lab3.cpp

```diff
@@ -0,0 +1,102 @@
+#include <mpi.h>
+#include <iostream>
+#include <string>
+#include <vector>
+
+int procId;
+int numProcs;
+
+struct student
+{
+    int id;
+    char name[10];
+    char surname[10];
+    student(int id, const std::string &name, const std::string &surname)
+    {
+        this->id = id;
+        memset(&this->name[0], 0, 10);
+        memset(&this->surname[0], 0, 10);
+        strcpy_s(this->name, name.c_str());
+        strcpy_s(this->surname, surname.c_str());
+    };
+    student() {};
+};
+
+inline bool operator==(const student& lhs, const student& rhs) 
+{
+    if (lhs.id == rhs.id && strcmp(lhs.name, rhs.name) == 0 && strcmp(lhs.surname, rhs.surname) == 0)
+        return true;
+    return false;
+}
+
+void testLab3(const MPI_Aint &studentType, const std::vector<student> &input, const student &searchedStud, bool expectedFound)
+{
+    MPI_Comm_rank(MPI_COMM_WORLD, &procId);
+    std::vector<student> localInput;
+
+    int procChunkSize = input.size() / numProcs;
+    localInput.resize(procChunkSize);
+    MPI_Scatter(&input[0], procChunkSize, studentType, &localInput[0], procChunkSize, studentType, 0, MPI_COMM_WORLD);
+
+    int localFound = false;
```

Comment:
you need to use `1` and `0` instead of `true` and `false` when working with `int`s.